src/doc/DocAnnotator.js

```diff
@@ -821,14 +773,15 @@ class DocAnnotator extends Annotator {
             this.highlighter.removeAllHighlights();
         }
 
-        const isCreateDialogVisible = this.createHighlightDialog && this.createHighlightDialog.isVisible;
+        const createDialog = this.plainHighlightEnabled
```

Comment:
I've noticed a little bit of duplicate code around this. Could you abstract this to a function?